src/storage.rs

```diff
@@ -840,65 +912,89 @@ impl StoreMut<String> for ProjectNextStorage {
+                                        },
+                                        DataType::Text => ProjectV2FieldValue {
+                                            text: new_value
+                                                .cast(&DataType::Text)
+                                                .ok()
+                                                .map(|v| v.into()),
+                                            ..Default::default()
+                                        },
+                                        _ => None?,
+                                    })
                                 }
+
+                                let Some(new_value_input) = into_update_input(&ty, new_value) else {
+                                    return Err((
+                                        self,
+                                        GlueSQLError::Value(ValueError::IncompatibleDataType {
+                                            data_type: ty,
+                                            value: new_value.clone(),
+                                        }),
+                                    ));
+                                };
+                                new_value_input
                             }
-                            FieldKind::Iteration {
-                                iterations,
-                                completed_iterations,
-                                ..
-                            } => {
-                                if let Some(opt) =
-                                    iterations.iter().find(|it| it.title == new_value_str)
-                                {
-                                    opt.id.clone()
-                                } else if let Some(it) = completed_iterations
-                                    .iter()
-                                    .find(|opt| opt.title == new_value_str)
-                                {
-                                    it.id.clone()
+                            FieldKind::SingleSelect(options) => {
+                                let new_str: String = new_value.into();
+                                if let Some(opt) = options.iter().find(|opt| opt.name == new_str) {
+                                    ProjectV2FieldValue {
+                                        single_select_option_id: Some(opt.id.to_owned()),
+                                        ..Default::default()
+                                    }
                                 } else {
                                     return Err((
                                         self,
                                         GlueSQLError::Value(ValueError::ImpossibleCast),
                                     ));
                                 }
                             }
+                            FieldKind::Iteration { .. } => {
+                                let new_str: String = new_value.into();
```

Comment:
I want Iteration title to id conversion for compatibility and convinience.
Could you implement it?

Reply:
𝑺𝑼𝑷𝑹𝑬𝑴𝑬